### PR TITLE
refactor(image): convert enzyme test to rtl

### PR DESCRIPTION
**Pull request by tomdavies73**
### Proposed behaviour

Converts enzyme test to RTL

### Current behaviour

Tests are currently written in enzyme

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/image/image.test.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  testStyledSystemPadding,
+} from "../../__spec_helper__/test-utils";
+import Image from "./image.component";
+
+testStyledSystemMargin((props) => <Image {...props} />);
+testStyledSystemLayout((props) => <Image {...props} />);
+testStyledSystemBackground((props) => <Image {...props} />);
+testStyledSystemPadding((props) => <Image {...props} />);
+
+it("renders", () => {
```
Comment (nuria1110) [inline src/components/image/image.test.tsx:16]:
```suggestion
test("renders", () => {
```
**comment**: To stay consistent with other test files I would use `test`

File: src/components/image/image.test.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  render(<Image data-role="image" />);
+
+  expect(screen.getByTestId("image")).toBeInTheDocument();
```
Comment (nuria1110) [inline src/components/image/image.test.tsx:19]:
```suggestion
  expect(screen.getByTestId("image")).toBeVisible();
```

**suggestion**: I think its better to use `toBeVisible` as it also asserts it visible to the user, same for the rest of the tests.

File: src/components/image/image.test.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+it("sets the 'alt' attribute correctly when a custom value is passed to the 'alt' prop", () => {
+  render(
+    <Image
+      data-role="image"
+      src="https://upload.wikimedia.org/wikipedia/en/6/6c/Heatposter.jpg"
+      alt="Heat poster"
+    />
+  );
+
+  expect(screen.getByRole("img")).toHaveAttribute("alt", "Heat poster");
```
Comment (nuria1110) [inline src/components/image/image.test.tsx:113]: **suggestion:** I think you could use an accessible name assertion for the alt text.  
```suggestion
  expect(screen.getByRole("img")).toHaveAccessibleName("Heat poster");
```

File: src/components/image/image.test.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+test("renders with the 'src' prop", () => {
+  render(
+    <Image
+      data-role="image"
+      src="https://upload.wikimedia.org/wikipedia/en/6/6c/Heatposter.jpg"
+      alt="Heat poster"
+    />
+  );
+
+  expect(screen.getByRole("img")).toBeInTheDocument();
```
Comment (nuria1110) [inline src/components/image/image.test.tsx:31]:
```suggestion
  expect(screen.getByRole("img")).toBeVisible();
```

File: src/components/image/image.test.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  render(
+    <Image
+      data-role="image"
+      src="https://upload.wikimedia.org/wikipedia/en/6/6c/Heatposter.jpg"
+      alt=""
+      decorative
+    />
+  );
+
+  expect(screen.getByRole("img")).toBeInTheDocument();
```
Comment (nuria1110) [inline src/components/image/image.test.tsx:44]:
```suggestion
  expect(screen.getByRole("img")).toBeVisible();
```

File: src/components/image/image.test.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+});
+
+test("renders with children", () => {
+  render(<Image>Children</Image>);
+
+  const children = screen.getByText("Children");
+  expect(children).toBeInTheDocument();
```
Comment (nuria1110) [inline src/components/image/image.test.tsx:51]:
```suggestion
  expect(children).toBeVisible();
```

**Review by nuria1110 — COMMENTED** [review]
Changes good good to me, though I would also use the `toBeVisible()` assertion for these tests:

**Review by DipperTheDan — COMMENTED** [review]
Looks OK to me. Just address the comments left by @nuria1110 and I'll happily approve 😄

**Review by nuria1110 — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**